### PR TITLE
kernel: properly name idle threads

### PR DESCRIPTION
**Pull request by tyrantlizard77**
These are now indexed by CPU.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

### CODE REVIEW

**Review by nashif — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

File: kernel/init.c [hunk 1]
```diff
@@ -279,11 +278,22 @@ void __weak main(void)
...
-static void init_idle_thread(struct k_thread *thread, k_thread_stack_t *stack)
+static void init_idle_thread(int i)
 {
+	struct k_thread *thread = &z_idle_threads[i];
+	k_thread_stack_t *stack = z_idle_stacks[i];
+
+#ifdef CONFIG_THREAD_NAME
+	char tname[8];
+
+	snprintk(tname, 8, "idle %02d", i);
```
Comment (stephanosio) [inline kernel/init.c]: `snprintk(tname, sizeof(tname), "idle %02d", i);`

File: kernel/init.c [hunk 2]
```diff
@@ -279,11 +278,22 @@ void __weak main(void)
...
 
 #if defined(CONFIG_MULTITHREADING)
-static void init_idle_thread(struct k_thread *thread, k_thread_stack_t *stack)
+static void init_idle_thread(int i)
 {
+	struct k_thread *thread = &z_idle_threads[i];
+	k_thread_stack_t *stack = z_idle_stacks[i];
+
+#ifdef CONFIG_THREAD_NAME
+	char tname[8];
```
Comment (stephanosio) [inline kernel/init.c]: Maybe `CONFIG_THREAD_MAX_NAME_LEN` here for maximum flexibility?

File: kernel/init.c (hunk 1 above)
Comment (stephanosio) [reply kernel/init.c]: Do we really need fixed 2-digit numbers? (e.g. `idle 00`, `idle 01`, ...) What if we have more than 99 cores? Wouldn't it be better to just name them `idle 0`, `idle 1`, ...?

Comment (andrewboie) [reply kernel/init.c]: > What if we have more than 99 cores

On a microcontroller OS? Really?

Comment (stephanosio) [reply kernel/init.c]: > On a microcontroller OS? Really?

Probably not, though with neural network being a thing nowadays, that probably isn't all that far-fetched (I mean we already have 64; I suspect 128 or even 1024 is not that far away).

File: kernel/init.c (hunk 2 above)
Comment (ioannisg) [reply kernel/init.c]: Agreed

**Review by ioannisg — APPROVED** [review]
Looks OK to me. +1 for using the Kconfig define instead of hard-coded 8, if possible

**Review by ceolin — APPROVED** [review]

**Pull request merged**